include/dlaf/matrix.h

```diff
@@ -103,6 +103,34 @@ class Matrix : public Matrix<const T, device> {
     return operator()(this->distribution().localTileIndex(index));
   }
 
+  /// Copy values from another matrix
+  ///
+  /// Given a matrix with the same geometries and distribution, this function submits tasks that will
+  /// perform the copy of each tile
+  template <class U, class = std::enable_if_t<std::is_same<T, std::remove_const_t<U>>::value>>
+  void copyFrom(Matrix<U, Device::CPU>& source) {
+    const auto& distribution = source.distribution();
+
+    // TODO check same size and blocksize
+    // TODO check equally distributed
+
+    const SizeType local_tile_rows = distribution.localNrTiles().rows();
+    const SizeType local_tile_cols = distribution.localNrTiles().cols();
+
+    for (SizeType j = 0; j < local_tile_cols; ++j) {
+      for (SizeType i = 0; i < local_tile_rows; ++i) {
+        hpx::dataflow(hpx::util::unwrapping([](auto&& tile_dst, auto&& tile_src) {
+                        for (SizeType j = 0; j < tile_src.size().cols(); ++j)
+                          for (SizeType i = 0; i < tile_src.size().rows(); ++i) {
+                            TileElementIndex index(i, j);
+                            tile_dst(index) = tile_src(index);
+                          }
+                      }),
+                      this->operator()(LocalTileIndex(i, j)), source.read(LocalTileIndex(i, j)));
```

Comment:
I'm not sure if this is an unpopular opinion but nesting function calls like that seems hard to read. How about this:

```
  using hpx::util::unwrapping;
  using hpx::dataflow;
  LocalTileIndex tile_idx(i, j);
  auto copy_f = unwrapping([](auto&& tile_dst, auto&& tile_src) {
    for (SizeType j = 0; j < tile_src.size().cols(); ++j)
      for (SizeType i = 0; i < tile_src.size().rows(); ++i) {
        TileElementIndex index(i, j);
        tile_dst(index) = tile_src(index);
      }
  });
  dataflow(copy_f, this->operator()(tile_idx), source.read(tile_idx));
```

Reply:
What about using the optimized [s,d,c,z]lacpy?

Reply:
> I'm not sure if this is an unpopular opinion but nesting function calls like that seems hard to read. How about this:
> 
> ```
>   using hpx::util::unwrapping;
>   using hpx::dataflow;
>   LocalTileIndex tile_idx(i, j);
>   auto copy_f = unwrapping([](auto&& tile_dst, auto&& tile_src) {
>     for (SizeType j = 0; j < tile_src.size().cols(); ++j)
>       for (SizeType i = 0; i < tile_src.size().rows(); ++i) {
>         TileElementIndex index(i, j);
>         tile_dst(index) = tile_src(index);
>       }
>   });
>   dataflow(copy_f, this->operator()(tile_idx), source.read(tile_idx));
> ```

@teonnik In general this is a good idea to improve readability!